src/client/mvt/feature-decoder.js

```diff
@@ -161,15 +162,102 @@ function _getPreClippedVertices(geom, mvtExtent) {
         let x = coord.x;
         let y = coord.y;
 
-        x = 2 * x / mvtExtent - 1;	
-        y = 2 * (1 - y / mvtExtent) - 1;	
+        x = 2 * x / mvtExtent - 1;
+        y = 2 * (1 - y / mvtExtent) - 1;
 
         return [x, y];
     });
 }
 
+function clipLine(line) {
+    // linestring clipping based on the Cohen-Sutherland algorithm
+    // input is a single linestring [point0, point1, ...]
+    // output is an array of flat linestrings:
+    // [[p0x, p0y, p1x, p1y, ...], ...]
+    let clippedLine = [];
+    const clippedLines = [];
+    function clipType(point) {
+        let type = 0;
+        for (let i = 0; i < 4; i++) {
+            type = type | (clippingEdges[i](point) ? 0 : (1 << i));
+        }
+        return type;
+    }
+    function intersect(point1, point2, type) {
+        for (let i = 0; i < 4; i++) {
+            const mask = 1 << i;
+            if (type & mask) {
+                const p = clippingEdgeIntersectFn[i](point1, point2);
+                type = clipType(p) & ~mask;
+                return [p, type];
+            }
+        }
+    }
+    let point0 = line[0];
+    let type0 = clipType(point0);
+    for (let i=1; i<line.length; ++i) {
+        let point1 = line[i];
+        let type1 = clipType(point1);
+        const nextType = type1;
+        const nextPoint = point1;
+
+        for(;;) {
```

Comment:
?

Reply:
What does `for(;;)` do 😳?

Reply:
It's an infinite loop 😁 I had used `white(true)` originally but lint didn't like it.
The loop repeats until the lines need not be clippend with any of the sides (right, top, left, bottom)

Reply:
:O didn't know about that.

I any case I think `while(true)` is more legible